plugins/query-insights/src/main/java/org/opensearch/plugin/insights/rules/action/top_queries/TopQueriesRequest.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugin.insights.rules.action.top_queries;
+
+import org.opensearch.action.support.nodes.BaseNodesRequest;
+import org.opensearch.common.annotation.PublicApi;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Locale;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * A request to get cluster/node level top queries information.
+ *
+ * @opensearch.internal
+ */
+@PublicApi(since = "1.0.0")
+public class TopQueriesRequest extends BaseNodesRequest<TopQueriesRequest> {
+
+    Metric metricType = Metric.LATENCY;
+
+    /**
+     * Constructor for TopQueriesRequest
+     *
+     * @param in A {@link StreamInput} object.
+     * @throws IOException if the stream cannot be deserialized.
+     */
+    public TopQueriesRequest(StreamInput in) throws IOException {
+        super(in);
+        setMetricType(in.readString());
+    }
+
+    /**
+     * Get top queries from nodes based on the nodes ids specified.
+     * If none are passed, cluster level top queries will be returned.
+     *
+     * @param nodesIds the nodeIds specified in the request
+     */
+    public TopQueriesRequest(String... nodesIds) {
```

Comment:
what are these `nodeIds`?  Coordinator node id where the request lands?

Reply:
For top N queries, each node maintains it's own priority queue for "top queries", Using nodeIds could provide a way to just get top queries from specified nodeIds.